### ISSUE TITLE
SPIKE: Investigate paths forward for NextJS Storefront project

### ISSUE THREAD

**Issue by brent-hoover** [reactioncommerce/example-storefront#812]
**This is a general Spike to do research to choose a path forward for Storefront:**

**Part 1:**

Look at the community built project at: https://github.com/tedraykov/airfit-storefront

1. What the quality of the project
2. Is the creator interested in this becoming an "official" project
3. What would be need to make it replace example storefront
4. Are there any licensing issues with taking another project that's not on our license.

Things we would want to consider:
1. How good is the test coverage?
2. Is it built in a way that anyone can use it as a jumping off point for their own implementation.
3. How complete is it?:
   a. You can browse categories that are dynamically created in OC
   b. You can add to cart and check out and get whatever shipping/method is configured (not hard-coding Stripe or anything like that)
   c. You can view order status for a logged in user or from the email that is sent.
   d. Can you use discount codes during checkout (something that doesn't exist in example-storefront)
   e. Generally compare it to example-storefront and make sure we aren't losing any significant features
   f. Does it display stock status
   g. Look for anything that's "hard-coded"

May want to look at Vercel's own implementation of the storefront with Shopify as well to compare.

Also as part of this process we should stand up a version of OC loaded with data (hopefully we have the sample data plugin done) using this storefront so everyone can view/test it.

**Part 2:** 

Look at the level of effort involved in creating a NextJS storefront from scratch and how that would compare with taking on the community project. This should include a very rough time estimate.

Basically we want to see the pros and cons of each approach and map out a way forward. This Spike explicitly **does not** include actually doing any of this work but is a research exercise.

The results of this should be a complete description of the analysis and a recommendation supported by the analysis. Also be prepared to do a Blog Post about the process and how analyzed the two.

This research would be part of putting together a PRD which I will start

**Comment by triomen** [comment]
Could you share the link of this project ? I didn't find it on github.
FYI, I found an "old" code to integrate OC with Vue Storefront, didn't try it https://github.com/pondigitalsolutions/rc-storefront-api

**Comment by brent-hoover** [comment]
@triomen https://github.com/tedraykov/airfit-storefront

**Comment by brent-hoover** [comment]
Closing this as we have decided to move forward with building our own NextJS Storefront implementation
